modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenTestCase.java

```diff
@@ -0,0 +1,591 @@
+    private List<Permission> userPermissions = new ArrayList<>();
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init(TestUserMode.TENANT_USER);
+
+        oAuth2Service = this.new OAuth2ServiceHelper();
+
+        restClient = new ActionsRestClient(serverURL, tenantInfo);
+        scim2RestClient = new SCIM2RestClient(serverURL, tenantInfo);
+        adminClient = new OauthAdminClient(backendURL, sessionCookie);
+
+        setSystemproperties();
+    }
+
+    // TODO: check what is wrong here
+//    @AfterClass(alwaysRun = true)
+//    public void atEnd() throws Exception {
+//        restClient.deleteV2Role(roleID);
+//        deleteApp(appId);
+//        scim2RestClient.deleteUser(userID);
+//        consumerKey = null;
+//        consumerSecret = null;
+//        appId = null;
+//        restClient.closeHttpClient();
+//    }
+
+    /**
+     * Provides data for testing registration pre-requisites.
+     * Each dataset consists of an external service name and its uri.
+     *
+     * @return Two-dimensional array containing pairs of external service names and its uris.
+     */
+    @DataProvider(name = "getExternalServiceInfo")
+    public Object[][] getExternalServiceInfo() {
+
+        return new Object[][]{
+                {"TestExternalService", "https://wso2is.free.beeceptor.com"}
+        };
+    }
+
+    @Test(groups = "wso2.is", description = "Registers an application with specified audience type, grant type. " +
+            "Creates a role, adds users and associates users with roles. Subscribe to necessary API resources.",
+            dataProvider = "getExternalServiceInfo")
+    private void testRegisterPreRequisites(String externalServiceName, String externalServiceURI) throws Exception {
+        appId = registerApplication(externalServiceName, externalServiceURI);
+        roleID = createRoles(appId);
+        userID = createUser();
+        assignRoleToUser(roleID, userID);
+    }
+
+    private String registerApplication(String externalServiceName, String externalServiceURI) throws Exception {
+        OAuthConsumerAppDTO applicationDTO = oAuth2Service.getBasicOAuthApp(OAuth2Constant.CALLBACK_URL);
+        applicationDTO.setTokenType(JWT);
+        applicationDTO.setGrantTypes(OAuth2Constant.OAUTH2_GRANT_TYPE_RESOURCE_OWNER);
+
+        AssociatedRolesConfig associatedRolesConfig = new AssociatedRolesConfig();
+        associatedRolesConfig.setAllowedAudience(APPLICATION_AUDIENCE);
+        ServiceProvider serviceProvider = null;
+        try {
+            serviceProvider = oAuth2Service.registerApplicationAudienceServiceProvider(applicationDTO, associatedRolesConfig);
+        } catch (Exception e) {
+            throw new RuntimeException("Error occurred while creating service provider: " + applicationDTO.getApplicationName());
+        }
+
+        applicationDTO = adminClient.getOAuthAppByName(serviceProvider.getApplicationName());
```

Comment:
Why not we depend on the rest api that this admin client. We can use rest assured and invoke rest apis to create the application as a pre-requisite for the test case

Reply:
Fixed with the commit 6e4aab9c2f07c6fa7ac9e7d13b2ea705224a25a5